src/deepl-translator.js

```diff
@@ -43,7 +62,8 @@ function translate(text, targetLanguage, sourceLanguage = 'auto') {
               : getTranslation(
                   paragraph,
                   targetLanguage,
-                  resolvedSourceLanguage || 'auto'
+                  resolvedSourceLanguage || 'auto',
+                  beginning
```

Comment:
Passing in a `beginning` into this method could be problematic if we have multiple paragraphs since each paragraph beginning would be overridden (as they are translated separately) and we probably don't want that. Equally unlikely would be that we want to replace only the very beginning of a large body of text. This method was meant to be simple and dumb. You pass in a chunk of text and get the best translation for it with formatting **somewhat** preserved, so I'd prefer not to add `beginning` override functionality to it as it could get confusing.

Instead, perhaps a little better solution would be to implement a more flexible, `translateWithAlternatives` method, which would handle **only** single paragraphs or even single phrases/sentences. An optional `beginning` override would make much more sense here and we could also have the method return a list of alternative translations.

Something like this:
```javascript
function translateWithAlternatives(
  text,
  targetLanguage,
  sourceLanguage = 'auto',
  beginning = ''
) {
  return validateInputs(text, targetLanguage, sourceLanguage)
    .then(valid =>
      getTranslation([text], targetLanguage, sourceLanguage, beginning)
    )
    .then(({ result: { source_lang, translations: [{ beams }] } }) => ({
      targetLanguage,
      resolvedSourceLanguage: source_lang,
      translation: beams[0].postprocessed_sentence,
      translationAlternatives: beams.map(
        ({ postprocessed_sentence }) => postprocessed_sentence
      ),
    }));
}
```

Reply:
This makes much sense to me! A `translateWithAlternatives` method gives access to more advanced features of the Deepl API and preserves the alternative translations provided in the response. I can implement that, but I'm not sure when. I'm in the last few weeks of my master thesis right now... ;)

Reply:
I'll merge your PR over the weekend and implement the flexible function. I have a few other things in my to do list, so perhaps I'll do them all and push a new release. Thanks again for your work and good luck with your master thesis!